renovate/utils/run-discovery.hs

```diff
@@ -24,22 +27,24 @@ main :: IO ()
 main = do
   [filename] <- getArgs
   bs <- BS.readFile filename
-  elf <- case parseElf bs of
-    Elf64Res warnings elf -> mapM_ print warnings >> return elf
-    _ -> die "not a 64-bit ELF file"
-  case elfMachine elf of
-    EM_PPC64 -> do
-      bin <- loadBinary @PPC defaultLoadOptions elf
-      entries <- toList <$> entryPoints bin
-      let pli = ppc64_linux_info bin
-      showDiscoveryInfo $ cfgFromAddrs pli (memoryImage bin) M.empty entries []
-    EM_X86_64 -> case resolveElfContents defaultLoadOptions elf of
-      Left e -> fail (show e)
-      Right (_, _, Nothing, _) -> fail "Couldn't work out entry point."
-      Right (warn, mem, Just entryPoint, _) -> do
-        mapM_ print warn
-        showDiscoveryInfo $ cfgFromAddrs x86_64_linux_info mem M.empty [entryPoint] []
-    _ -> fail "only X86 and PPC64 supported for now"
+  case EE.decodeElfHeaderInfo bs of
+    Left (off, err) -> die ("Error decoding ELF file at " ++ show off ++ ": " ++ err)
+    Right (EE.SomeElf e) ->
+      case EE.headerClass (EE.header e) of
+        EE.ELFCLASS32 -> die "32 bit binaries not supported"
+        EE.ELFCLASS64 ->
+          case EE.headerMachine (EE.header e) of
+            EM_PPC64 -> do
+              bin <- loadBinary @PPC defaultLoadOptions e
+              entries <- toList <$> entryPoints bin
+              let pli = ppc64_linux_info bin
+              showDiscoveryInfo $ cfgFromAddrs pli (memoryImage bin) M.empty entries []
+            EM_X86_64 -> do
+              bin <- loadBinary @X86_64 defaultLoadOptions e
+              entries <- toList <$> entryPoints bin
+              showDiscoveryInfo (cfgFromAddrs x86_64_linux_info (memoryImage bin) M.empty entries [])
+            _ -> fail "only X86 and PPC64 supported for now"
```

Comment:
Good place to use `panic`

Reply:
There are a couple of `die` calls above.   Is `die` different than `fail` here? Should `die` call `panic`?

Reply:
I'll get them all unified eventually - I was trying to not go too off the rails on this PR